submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireFilesService.kt

```diff
@@ -78,24 +80,25 @@ class FireFilesService(
         val files = client.findByMd5(file.md5)
 
         val byPath = files.firstOrNull { it.filesystemEntry?.path == expectedPath }
-        if (byPath != null) return asFireFile(file, byPath.fireOid)
+        if (byPath != null) return asFireFile(file, byPath.fireOid, byPath.filesystemEntry!!.path!!)
 
         val noPath = files.firstOrNull { it.filesystemEntry?.path == null }
-        if (noPath != null) return setMetadata(noPath.fireOid, file, expectedPath)
+        if (noPath != null) return setMetadata(file, noPath, expectedPath)
 
         val saved = client.save(fallbackFile(), file.md5, file.size)
-        return setMetadata(saved.fireOid, file, expectedPath)
+        return setMetadata(file, saved, expectedPath)
     }
 
-    private fun setMetadata(fireOid: String, file: ExtFile, expectedPath: String): FireFile {
-        client.setPath(fireOid, expectedPath)
-        return asFireFile(file, fireOid)
+    private fun setMetadata(file: ExtFile, fireFile: FireApiFile, expectedPath: String): FireFile {
```

Comment:
I don't understand the benefit of this change 😕

Reply:
totalle agree